src/blenderbim/blenderbim/bim/module/material/operator.py

```diff
@@ -152,13 +157,16 @@ def execute(self, context):
         return IfcStore.execute_ifc_operator(self, context)
 
     def _execute(self, context):
-        obj = bpy.data.objects.get(self.obj) if self.obj else context.active_object
         self.file = IfcStore.get_file()
-        ifcopenshell.api.run(
-            "material.unassign_material",
-            self.file,
-            **{"product": self.file.by_id(obj.BIMObjectProperties.ifc_definition_id)},
-        )
+        objs = [bpy.data.objects.get(self.obj)] if self.obj else context.selected_objects
+        for obj in objs:                       
+            if not obj.BIMObjectProperties.ifc_definition_id:
+                continue
+            ifcopenshell.api.run(
+                "material.unassign_material",
+                self.file,
+                **{"product": self.file.by_id(obj.BIMObjectProperties.ifc_definition_id)},
+            )
```

Comment:
Same - please revert, sorry :(

Reply:
Haha no problem I would rather not break everything when trying to streamline things